File: src/clj/at/ac/tuwien/complang/distributor/worker.clj
(ns at.ac.tuwien.complang.distributor.worker
  (:import [at.ac.tuwien.complang.distributor Worker NotImplementedException]))

(defn worker-server [funcs]
  (reify
   Worker
   (compute [this fun-name args]
	    (let [fun (funcs fun-name)]
	      (if fun
		(apply fun args)
		(throw (NotImplementedException.)))))))

File: src/java/at/ac/tuwien/complang/distributor/Worker.java
package at.ac.tuwien.complang.distributor;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.io.Serializable;

public interface Worker extends Remote {
	Serializable compute (Serializable function, Serializable args) throws RemoteException, NotImplementedException;
}
